### DISCUSSION TITLE
matching up application status with Stripe

### DISCUSSION THREAD

**Discussion by dvodvo** [pay-rails/pay#701] [General] [answered]
In the case of a subscription, there are multiple avenues of operation if a subsequent payment experiences insufficient funds or whether the subscriber cancelled directly with the payment processor.

Grace period is opaque.  Assumption:
`@user.payment_processor.subscription.on_grace_period? ` implies, a configuration with the payment processor to allow daily attempts to re-try the payment.  Yes?  

If so, the nomenklature of 'grace' does not exist within Stripe (that string does not come up in any searches); where is that set?

2nd assumption, it is via a job task to query subscriptions and take appropriate actions with respect to application content access, via:
`@user.payment_processor.subscribed?`to gather the payment status.  Yes?

Stripe works with a product_pricing_id.  If a user has multiple subscriptions:  
`@user.payment_processor.subscribed?(name: "default", processor_plan: "monthly")`
I would assume a query via the product_pricing_id.  Not clear to this observer.

**Comment by excid3** [comment]
Grace period is when a user has canceled their subscription but still has access until the end of the period they paid for.

**Reply by dvodvo** [reply]
And this is a cancel that is initiated by the app and/or via stripe (assuming the client has logged with them) ?

**Reply by excid3** [reply]
It doesn't matter how it was canceled. Just subscriptions marked to be canceled at end of period.

**Comment by excid3** [comment, marked as answer]
Yes, you would use the price_id as the value for processor_plan. Stripe renamed Plans to Price and we support more than just Stripe, so it will stay as "plan".
